test/e2e/all_in_one_test.go

```diff
@@ -355,7 +355,13 @@ func TestDeployAllInOneDBLESSMultiGW(t *testing.T) {
 		}
 		podList, err = env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
 		require.NoError(t, err)
-		return len(podList.Items) == 3
+
+		readyCount := lo.CountBy(podList.Items, func(item corev1.Pod) bool {
+			return lo.CountBy(item.Status.Conditions, func(item corev1.PodCondition) bool {
+				return item.Type == corev1.PodReady && item.Status == corev1.ConditionTrue
+			}) == 1
+		})
+		return readyCount == 3
```

Comment:
While we're at it do you mind extract the `3` into a `const` (and using it in line  345 as well)?

Reply:
[1ea847e](https://github.com/Kong/kubernetes-ingress-controller/pull/3566/commits/1ea847e0fadda418a172de417b05fcb6e7b62a0d)